SportsHub/UnitTests/Services/TestUserService.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SportsHub.AppService.Services;
+using SportsHub.DAL.Data;
+using SportsHub.DAL.UOW;
+using SportsHub.Domain.UOW;
+using Xunit;
+
+namespace UnitTests.Services;
+
+public class TestUserService : IDisposable
+{
+    private readonly ApplicationDbContext _context;
+    private readonly IUnitOfWork _unitOfWork;
+
+    public TestUserService()
+    {
+        var options = new DbContextOptionsBuilder<ApplicationDbContext>()
+            .UseInMemoryDatabase(databaseName: Guid.NewGuid().ToString())
+            .Options;
```

Comment:
let's discuss if we need to create a DB here

Reply:
I have created it because the UnitOfWotk needs a database . In my test coverage i saw that with this approach i covered the database too .

Reply:
If you want to cover a Database layer as well, then add a test project for the Store layer and add tests. That is not the responsibility of service tests. Unit means that we test only small peace of the functionality

Reply:
But i cant mock the UOF and it is needed for the UserService . So this is the way that i have found to test it . Can you point me how to change this and test it without inline database for the UOF

Reply:
you can mock it, Mock<IunitOfWork> mock= new Mock<IUnitOfWork>(). and then use the setup for getting data